src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,126 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int capacity = 10;
+    private static final double COEFFICIENT_GROWING = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < capacity) {
+            array[size] = value;
+            size++;
+        } else {
+            grow(capacity, value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index <= size && size < capacity && index > -1) {
+            if (capacity - 1 > size) {
+                T[] arrayWithNewCapacity = (T[]) new Object[capacity + 1];
+                System.arraycopy(array, index, arrayWithNewCapacity, index + 1, size - index);
+                arrayWithNewCapacity[index] = value;
+                System.arraycopy(array, 0, arrayWithNewCapacity, 0, index);
+                array = arrayWithNewCapacity;
+                size++;
+            } else {
+                grow(capacity, value, index);
+            }
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Can`t add element with this index.");
+        }
```

Comment:
lets check size first - call grow() method if needed
and then just move the part after the index 1 cell to the right
and set value on index that you need